sdk/java/pulumi/src/test/java/com/pulumi/deployment/MocksTest.java

```diff
@@ -112,10 +116,9 @@ void testStack() {
                 .setMocks(new MyMocks())
                 .setSpyGlobalInstance();
 
-        var result = mock.tryTestAsync(MyStack::init).join();
-        var stack = MyStack.of(result);
-        var ip = OutputTests.waitFor(stack.publicIp).getValueNullable();
-        assertThat(ip).isEqualTo("203.0.113.12");
+        var result = mock.runTestAsync(MocksTest::myStack).join();
+        var publicIp = waitForValue(result.stackOutput("publicIp", String.class));
```

Comment:
Ok here we need to tell it it's a `String.class`, but does not seem too bad.

Reply:
yes, this avoids nasty bugs, Java serializers usually use this kind of API